src/Hedgehog/Plutus/TestSingleScript.hs

```diff
@@ -106,8 +127,17 @@ txRunScript
         ) =>
         Proxy era ->
         Ledger.PParams era ->
+        Ledger.Tx era ->
         Maybe Plutus.EvaluationError
-      go _ params =
+      go _ params coreTx = do
+        let utxoMap = first (`Fork.TxIn` Nothing) <$> Map.toList utxos
+        -- TODO is this Nothing correct?
```

Comment:
It is not.

The UTxO type/parameter is _very_badly named. It is a assocmap of **the inputs of the transaction** resolved to their outputs. the txins come from the tx, the txouts are resolved from mockUtxos.